server/manager/src/main/java/org/apache/accumulo/master/TabletGroupWatcher.java

```diff
@@ -230,117 +232,25 @@ public void run() {
           sendChopRequest(mergeStats.getMergeInfo(), state, tls);
           sendSplitRequest(mergeStats.getMergeInfo(), state, tls);
 
-          // Always follow through with assignments
-          if (state == TabletState.ASSIGNED) {
-            goal = TabletGoalState.HOSTED;
-          }
+          // Always follow through with assignments will give AssignedTabletHostedGoal
```

Comment:
With this design, the "Always follow through with assignments" check would not need to exist, they would have this logic built into the types. For example, ```AssignedTabletHostedGoal``` is the only class that has a current state of ```ASSIGNED```.